### DISCUSSION TITLE
CSP config for custom domain

### DISCUSSION THREAD

**Discussion by alrick** [directus/directus#15116] [Archive]
Hey there,

I run Directus self-hosted with a custom domain.
It seems that with the default variables, I have some CSP errors.

I found some information in the issues but can't find a clear explanation on how to setup my installation, can someone help me with the default CSP config for custom domains ? Or redirect me to some documentation ?

Thanks a lot

**Comment by rijkvanzanten** [comment]
What CSP errors are you seeing?

**Reply by alrick** [reply]
At the moment I see only 
`Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”).`

But the netmaster disabled some of the rules they set on the load balancer because it broke the app.

Here is the full list of errors when they enable all rules : 
![eddb-csp](https://user-images.githubusercontent.com/1086355/185159983-27e7b957-aa3c-4386-817f-735adea0f251.png)

**Reply by rijkvanzanten** [reply]
Looks like the default src CSP header was adjusted in the environment variables. Make sure that directus is able to execute scripts inline and from data:/blob: sources. These are the defaults:

https://github.com/directus/directus/blob/608596f382cc20038114424de421348437e61dff/api/src/app.ts#L103-L124

**Reply by alrick** [reply]
I'll give this information to the netmaster and come back to you if needed.
Thank you for your help.
